src/components/HomePage/SearchArea/SearchArea.spec.js

```diff
@@ -55,6 +55,128 @@ describe("SearchArea", () => {
             expect(screen.getByRole("button", { name: "Search" })).toBeInTheDocument();
             expect(screen.getByRole("button", { name: "Toggle Advanced Search" })).toBeInTheDocument();
         });
+        it("should render Show All text when there is no filters or search bar has no value", () => {
+            const searchArea = renderWithStoreAndTheme(
+                <RouteWrappedContent>
+                    <SearchArea
+                        searchValue={""}
+                        jobType={INITIAL_JOB_TYPE}
+                        jobDuration={[null, null]}
+                        filterJobDuration={false}
+                        offers={[]}
+                        fields={[]}
+                        technologies={[]}
+                        setShowJobDurationSlider={() => { }}
+                        setTechs={() => { }}
+                        setJobDuration={() => { }}
+                        setFields={() => { }}
+                        setJobType={() => { }}
+                        setSearchValue={() => { }}
```

Comment:
Perhaps one thing you could do is create a separate test component that passes default props to SearchArea, kinda like this:
```js
const SearchAreaWithProps = (props = {<here you put all the props with default values or semthing like that>}) => <SearchArea {...props} />;
```

Then you would just need to overwrite the prop(s) that you wanted to test.

I do not recall off the top of my head if this is syntatically correct but the idea is there.